### PR TITLE
docs: add testing guide for Tegel and link in README

### PR DESCRIPTION
**Pull request by ckrook**
## **Describe pull-request**  
I have written documentation for how we do testing within Tegel. Please have a look at this and let me know what you think.

## **Issue Linking:**  
[CDEP-2939](https://tegel.atlassian.net/browse/CDEP-2939)

[CDEP-2939]: https://tegel.atlassian.net/browse/CDEP-2939?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mistermalm — APPROVED** [review]
nice!

File: docs/TESTING_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+This guide provides developers with best practices and guidelines on how and when to write unit tests for components within our design system. Our aim is to ensure that each component is reliable, maintainable, and functions as intended.
+
+---
+
+## How to Unit Test Components
+
+### Testing Setup
+
+- We use **Playwright** for unit testing.
+- Each component should have its own test file located in the same folder as the component (`componentName.e2e.ts`).
```
Comment (theJohnnyMe) [inline docs/TESTING_GUIDE.md:14]:
```suggestion
- Each component should have its test file located in the folder of the component (`../components/button/test/button.e2e.ts`).
```

Comment (theJohnnyMe) [reply docs/TESTING_GUIDE.md:14]: Also, we used folders to specify what kind of test is conducted, for example: disabled, danger etc... so usually more complex component have a subfolders that indicate specific test area.
Take a look on button component in code and see what is the best way to present this to the user.

File: README.md [hunk 2]
```diff
@@ -72,6 +72,10 @@ Get in touch with the team and the community:
 
 For users that can't access our CDN there is a [script](https://github.com/scania-digital-design-system/tegel/blob/develop/packages/core/static-assets-copying.md) for copying over static assets(fonts, logos).
 
+## Testing Guide for Tegel
+
+[This](./docs/TESTING_GUIDE.md) guide provides developers with best practices and guidelines on how and when to write unit tests for components within our design system. Our aim is to ensure that each component is reliable, maintainable, and functions as intended.
```
Comment (timrombergjakobsson) [inline README.md:77]: Just a thought, but maybe we should not have the same text in the link here pointing to the document?
So maybe just something like this suffice:

```suggestion
Refer to our [Testing Guide](./docs/TESTING_GUIDE.md) for best practices on writing unit tests for components in our design system.

[This](./docs/TESTING_GUIDE.md) guide provides developers with best practices and guidelines on how and when to write unit tests for components within our design system. Our aim is to ensure that each component is reliable, maintainable, and functions as intended.
```
?

Comment (ckrook) [reply README.md:77]: I think this makes a lot of since 👍

File: docs/TESTING_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
```
Comment (theJohnnyMe) [inline docs/TESTING_GUIDE.md:14]:
```suggestion
- Each component should have its own test file located in the same folder as the component (`../components/componentName/test/componentName.e2e.ts`).
```

**Comment by timrombergjakobsson** [comment]
Looks good, but seems as though some additional files where added that should not be in this PR?

**Comment by theJohnnyMe** [comment]
> Looks good, but seems as though some additional files where added that should not be in this PR?

I think rebase will sort it out.

**Review by theJohnnyMe — APPROVED** [review]
Thanks for sorting this out Charles! 
LGTM!

PS: I did rebase and force push so now we see only changes done in this PR. @timrombergjakobsson

**Pull request merged**